### PR TITLE
New integration test for record+verify

### PR DESCRIPTION
**Pull request by drevell**
We had tests for golden-test recording and verification separately, but not together. We need to verify that the directory created by the `record` command can be correctly handled by the `verify` command. I found a bug in my code by running CLI commands by hand that probably should have been covered by a test suite. This closes the gap.

### CODE REVIEW

File: templates/commands/goldentest/record_verify_test.go [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+// TestRecordVerify tests that the output of the record command can be verified by the verify
+// command.
+func TestRecordVerify(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name            string
+		testNames       []string
+		templateContent map[string]string
+		messWith        func(*testing.T, string)
```
Comment (capri-xiyue) [inline templates/commands/goldentest/record_verify_test.go:35]: what's the meaning of messWith?

File: templates/commands/goldentest/record_verify_test.go [hunk 2]
```diff
@@ -0,0 +1,148 @@
+// Copyright 2023 The Authors (see AUTHORS file)
```
Comment (capri-xiyue) [inline templates/commands/goldentest/record_verify_test.go:1]: nits: 2024

**Review by capri-xiyue — DISMISSED** [review]

File: templates/commands/goldentest/record_verify_test.go [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+	"github.com/abcxyz/abc/templates/common"
+	"github.com/abcxyz/pkg/logging"
+	"github.com/abcxyz/pkg/testutil"
+)
+
+// TestRecordVerify tests that the output of the record command can be verified by the verify
+// command.
+func TestRecordVerify(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
```
Comment (sethvargo) [inline templates/commands/goldentest/record_verify_test.go:31]:
```suggestion
	t.Parallel()

	cases := []struct {
```

File: templates/commands/goldentest/record_verify_test.go (hunk 1 above)
Comment (drevell) [reply templates/commands/goldentest/record_verify_test.go]: Added a comment.

File: templates/commands/goldentest/record_verify_test.go (hunk 3 above)
Comment (drevell) [reply templates/commands/goldentest/record_verify_test.go]: Done.

**Review by capri-xiyue — APPROVED** [review]

**Pull request merged**